File: packages/deterministic-mapping/src/mappings/sleep.ts
import { DIMENSIONS, DOMAINS, INTENTION_TYPES, CanonicalFact, MappingResult } from '../types';

export function mapSleepFacts(facts: CanonicalFact[]): MappingResult | null {
  const sleepFact = facts.find((f) => f.dimension === DIMENSIONS.SLEEP_STATE);
  if (!sleepFact) return null;

  return {
    domain: DOMAINS.SLEEP,
    type: INTENTION_TYPES.SLEEP_LOG,
    attributes: { state: sleepFact.value },
    confidence: sleepFact.confidence,
  };
}

File: packages/mcp-intent-gateway/src/mcp_intent_gateway/mapping/sleep.py
"""Sleep mapping function."""

from typing import Any

from ..domain.constants import Dimension, Domain, IntentionType
from ..models.canonical_fact import CanonicalFact


def map_sleep_facts(facts: list[CanonicalFact]) -> dict[str, Any] | None:
    """
    Map sleep-related facts to a MappingResult.

    Args:
        facts: List of canonical facts to process

    Returns:
        Mapping result dict or None if no sleep facts found
    """
    sleep_fact = next(
        (f for f in facts if f.dimension == Dimension.SLEEP_STATE),
        None,
    )

    if not sleep_fact:
        return None

    return {
        "domain": Domain.SLEEP,
        "type": IntentionType.SLEEP_LOG,
        "attributes": {"state": sleep_fact.value},
        "confidence": sleep_fact.confidence,
    }
